dotCMS/src/main/java/com/dotmarketing/portlets/cmsmaintenance/util/AssetFileNameFilter.java

```diff
@@ -1,62 +1,53 @@
 package com.dotmarketing.portlets.cmsmaintenance.util;
 
+import com.google.common.collect.ImmutableSet;
 import java.io.File;
 import java.io.FileFilter;
 
 import com.dotmarketing.util.Config;
 import com.dotmarketing.util.Logger;
 import com.liferay.util.FileUtil;
+import java.util.Set;
 
 public class AssetFileNameFilter implements FileFilter {
 
+    private static final Set<String> EXCLUDE_FOLDERS_LIST = ImmutableSet
+            .of("license", "bundles", "tmp_upload",
+                    "timemachine", "integrity", "server", "dotGenerated");
+
 	@Override
 	public boolean accept(File dir) {
 		if(dir ==null){
 			return false;
 		}
 
-
 		if(dir.getAbsolutePath().contains("dotGenerated") ){
 			return false;
 		}
+		
 		String name = dir.getName();
-		String osname = System.getProperty("os.name");
-		String[] path = null;
+		String osName = System.getProperty("os.name");
+		String[] path;
 
-		if (osname.startsWith("Windows"))
+		if (osName.startsWith("Windows"))
 			path = dir.getAbsolutePath().split("\\\\");
 		else
 			path = dir.getAbsolutePath().split(File.separator);
 
-
 		String[] test = new String[0];
-		String assetPath=null;
+
+		String assetPath;
         try {
         	assetPath = Config.getStringProperty("ASSET_REAL_PATH", FileUtil.getRealPath(Config.getStringProperty("ASSET_PATH")));
         	test = new File(assetPath).getAbsolutePath().split(File.separator);
         } catch (Exception e) {
         	Logger.debug(this.getClass(), e.getMessage());
         }
-        if(test.length +1 == path.length){
-
-			if( "license".equals(name)){
-				return false;
-			}
-			else if( "bundles".equals(name)){
-				return false;
-			}
-			else if( "tmp_upload".equals(name)){
-				return false;
-			}
-			else if( name.startsWith(".")){
-				return false;
-	        }
-			else if(name.equals("timemachine")) {
-			    return false;
-			}
+        if(test.length + 1 == path.length &&
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)